### ISSUE TITLE
RDF-native interpretation/profile

### ISSUE THREAD

**Issue by VladimirAlexiev** [admin-shell-io/aas-specs#384]
@sebbader in #45 on Sep 1, 2022:
> We are also currently working on a way more RDF-native interpretation/profile in addition to the official RDF schema,

Sebastian, can you update us on the status of that effort? 
There are a number of issues that call for a better RDF representation, and can be the basis for discussion:
- [ ] #45 
- [ ] #284
- [ ] #379
- [ ] #380
- [ ] #382
- [ ] #383

cc @mristin @mhrimaz @kenwenzel @arnoweiss

**Comment by kenwenzel** [comment]
Hi @VladimirAlexiev,

thank you for bringing this up.

We are also currently discussing:
- https://github.com/admin-shell-io/aas-specs/issues/43
- https://github.com/admin-shell-io/aas-specs/issues/166

**Comment by mristin** [comment]
Thanks, @VladimirAlexiev !

I do not use RDF myself, so I can't contribute much.

Pleaee mind that there is a perennial problem with the maintainability. The schemas were originally written by hand which turned out to be unmaintainable over the course of time. We ended up generating the schemas programmatically with aas-core using the formalized meta-model as a single-point-of-truth both for the schemas as well as the SDKs.

Many of the features you suggested are probably not possible right now due to the limitations in aas-core, but this should not be the blocker! If there is enough commitment from the volunteers, it probably make sense to generate the schema by hand, and then write a script to patch the additional features on top.

Alternatively, we could build in the features you need in aas-core. This would require additinal developers, which the aas-core maintainers lack at the moment. Of course, we are open for collaboration, so if anybody wants to work on that, I will gladly mentor.

**Comment by VladimirAlexiev** [comment]
@kenwenzel thanks! I added the two to the checkbox list in Description, and I'll see if I have anything to add to the discussion.

**Comment by VladimirAlexiev** [comment]
@egekorkan do you use AAS?

**Comment by egekorkan** [comment]
> @egekorkan do you use AAS?

Yes. We also have integration of WoT into AAS via [the Asset Interfaces Submodel](https://industrialdigitaltwin.org/en/wp-content/uploads/sites/2/2024/01/IDTA-02017-1-0_Submodel_Asset-Interfaces-Description.pdf) where we also have [converters](https://github.com/eclipse-thingweb/node-wot/tree/master/packages/td-tools/src/util) for it.

**Comment by VladimirAlexiev** [comment]
@sebbader-sap 
> My main problem is that the whole RDF (in particular JSON-LD) needs more "love" (aka. "time"), which is currently the limiting factor. I know that the "WG Ontologies" also has a bunch of improvement proposals in their pipeline, which all together would create a sound picture.
If we can combine all these different activities, maybe in late April or so, I can certainly help too.
It's an IDTA-internal group, I don't think that there is a public space. Is Ontotext an IDTA member? If so, you can join them.

Hi Seb!
- I'll enquire about becoming a member.
- But who can we talk to in order to get some encouragement (potential use, business prospects, etc)? We're not a big company, and trailblazing is an expensive activity...
- According to others here, AAS RDF has almost not been used to date. I know of some use by Mondragon and Festo (eg robotic cases) of GraphDB on a trial basis, and Siemens people use RDF for WoT/AAS conversions. Are there other examples?
- Does this WG have a home page, who are the members?
- If the process of this WG is closed (eg doesn't use github, PRs, test files, etc) then we'd be much less interested.

Thanks for your replies!

**Comment by kenwenzel** [comment]
@VladimirAlexiev You can also take part in the WG as a non-member but you wont have access to the IDTA's Teams and Sharepoint.
The process of the WG is not closed but currently we are just using files on IDTA's Sharepoint. This is something that can easily be changed if requested.

**Comment by VladimirAlexiev** [comment]
(I added two minor issues to the list on top.)

Thanks @kenwenzel! 
What is the "RDF-native interpretation/profile" that @sebbader-sap refers to? Is it "only" documents, or also machine-readable files?

@mristin wrote something important above:
> it probably make sense to generate the schema by hand, and then write a script to patch the additional features on top.

You're most competent to make this judgement, but the JSONLD context gives us a lot of flexibility that we should leverage to cast the JSON to a reasonable RDF; only then resort to patches. This said, I'll help with the patches!
